r/neotoma_tutorial.Rmd

```diff
@@ -396,32 +489,24 @@ extinct_tol <- extinct_tol[!is.na(extinct_tol)]
 
 ### Make data frame to merge with 
 
-[TODO] THis is an old way of making a dataframe.  Is there a more efficient way? This is super unreadable and confusing code. Maybe skip this part and upload the dataframe after?
+[TODO] Switch from read/write to using this if you think this approach is ok.
 
-```{r eval=FALSE}
-extinct_tol_list <- list()
-for (i in 1:length(names(extinct_tol))) {
-  taxon.name <- names(extinct_tol)[i]
+```{r, eval=FALSE}
+helper <- function(x,y){
 
-  x <- extinct_tol[[i]] %>% 
-    filter(rank %in% c("order", "family", "genus")) %>%
-    select(-id)
+  xNice = x %>% filter(rank %in% c("order", "family", "genus")) %>%
+    select(-id) %>% complete(rank = c("order","family","genus"))
   
-  if(dim(x)[1] != 3) { #probably need a better check here.
-    missing <- setdiff(c("order", "family", "genus"), x$rank)
-    add <- data.frame(name = rep(NA, length(missing)), rank=missing)
-    x <- rbind(x, add)
-  }
-  
-  y <- c(taxon.name, as.character(x$name))
-  names(y) <- c("taxon.name", "order", "family", "genus")
-  extinct_tol_list[[i]] <- y
-
+  y <- c(y, as.character(xNice$name))
+  return(y)
 }
 
-
+## see why I'm removing 74 and 200 below
+## [TODO] Check that these numbers stay the same after we deal with the discrepancy in manual selection key
```

Comment:
I'll have to double check this once we figure out the manual selection discrepancy.